setup.py

```diff
@@ -4,19 +4,27 @@
 import shutil
 import glob
 
+from Cython.Compiler.Version import version as cython_version
+from packaging.version import Version
+
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
 from Cython.Build import cythonize
 
 CYTHON_COVERAGE = os.environ.get("CYTHON_COVERAGE", False)
 define_macros = []
-compiler_directives = {}
+compiler_directives = dict()
+compiler_tenv = dict()
 if CYTHON_COVERAGE:
     print("CYTHON_COVERAGE is set. Enabling Cython coverage support.")
     define_macros.append(("CYTHON_TRACE_NOGIL", "1"))
     compiler_directives["linetrace"] = True
-
+if Version(cython_version) >= Version("3.1.0a0"):
```

Comment:
I'd just say newer than 3.1.0 - approximately no one is going to be using a prerelease Cython release now that 3.1.0 is out.